qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -767,3 +773,130 @@ def ds_with_vals(self, dataset):
+    expected_names['that'] = ['that', 'this_setpoint', 'that_setpoint']
+    expected_shapes = {}
+    if 'array' in types:
+        expected_shapes['this'] = [(5, 3), (5, 3)]
+        expected_shapes['that'] = [(5, 3), (5, 3)]
+    else:
+        expected_shapes['this'] = [(15,), (15,)]
+        expected_shapes['that'] = [(15,), (15,)]
+    parameter_test_helper(multi_dataset,
+                          input_names,
+                          expected_names,
+                          expected_shapes)
+
+
+def test_get_array_in_scalar_param_data(array_in_scalar_dataset):
+    paramspecs = array_in_scalar_dataset.paramspecs
+    types = [param.type for param in paramspecs.values()]
+
+    input_names = ['testparameter']
+
+    expected_names = {}
+    expected_names['testparameter'] = ['testparameter', 'scalarparam',
+                                       'this_setpoint']
+    expected_shapes = {}
+    if 'array' in types:
+        expected_shapes['testparameter'] = [(9, 5), (9, 5)]
+    else:
+        expected_shapes['testparameter'] = [(45,), (45,)]
+    parameter_test_helper(array_in_scalar_dataset,
+                          input_names,
+                          expected_names,
+                          expected_shapes)
+
+
+def test_get_array_in_str_param_data(array_in_str_dataset):
+    paramspecs = array_in_str_dataset.paramspecs
+    types = [param.type for param in paramspecs.values()]
+
+    input_names = ['testparameter']
+
+    expected_names = {}
+    expected_names['testparameter'] = ['testparameter', 'textparam',
+                                       'this_setpoint']
+    expected_shapes = {}
+    if 'array' in types:
+        expected_shapes['testparameter'] = [(3, 5), (3, 5)]
+    else:
+        expected_shapes['testparameter'] = [(15,), (15,)]
+    parameter_test_helper(array_in_str_dataset,
+                          input_names,
+                          expected_names,
+                          expected_shapes)
+
+
+def test_get_parameter_data_independent_parameters(standalone_parameters_dataset):
+    ds = standalone_parameters_dataset
+    params = get_non_dependencies(ds.conn, ds.run_id)
+
+    expected_toplevel_params = ['param_1', 'param_2', 'param_3']
+    assert params == expected_toplevel_params
+
+    expected_names = {}
+    expected_names['param_1'] = []
+    expected_names['param_2'] = []
+    expected_names['param_3'] = ['param_0']
+
+    expected_shapes = {}
+    expected_shapes['param_1'] = [(10 ** 3,)]
+    expected_shapes['param_2'] = [(10 ** 3,)]
+    expected_shapes['param_3'] = [(10**3, )]*2
+
+    parameter_test_helper(ds,
+                          expected_toplevel_params,
+                          expected_names,
+                          expected_shapes)
+
+
+def parameter_test_helper(ds, toplevel_names, expected_names, expected_shapes):
+    data = ds.get_parameter_data(*toplevel_names)
+    all_data = ds.get_parameter_data()
+
+    assert data.keys() == all_data.keys()
```

Comment:
this is a weird helper function because it only works for the cases where `toplevel_names` for a given `ds` happen to coincide with the names of non-dependencies (which are standalone + dependent parameters). I suggest to make the helper more generic (and perhaps also add tests where `set(data.keys()).is_subset_of(set(all_data.keys()))`?). Otherwise, this implicit assumption should be mentioned in comments.